### PR TITLE
LibGUI/FileIconProvider: Add config file

### PR DESCRIPTION
**Pull request by linusg**
Also add filetype-json icons.

### CODE REVIEW

File: Base/etc/FileIconProvider.ini [hunk 1]
```diff
@@ -16,3 +16,22 @@ python=*.py
 sound=*.wav
 spreadsheet=*.sheets
 text=*.txt
+
+[Overrides]
```
Comment (awesomekling) [inline Base/etc/FileIconProvider.ini:21]: This feels very hackish. How about we just rename the files to `*.json` instead? :D

Comment (linusg) [reply Base/etc/FileIconProvider.ini:21]: Indeed, it is quite hackish... but can we really do that without breaking stuff, same for the "text" files in `/proc`? `/proc/uptime.txt`? :thinking:

This is just a nice-to-have ofc, we can also leave it at the generic file icon.

Comment (awesomekling) [reply Base/etc/FileIconProvider.ini:21]: Sure, it'll break some hard-coded paths here & there, but we can fix those :)
Everything the light touches is our kingdom!

Comment (alimpfard) [reply Base/etc/FileIconProvider.ini]: All I see here is `br /proc/all.html`

**Pull request merged**